xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -90,6 +95,20 @@ public WeightedRoundRobinLoadBalancer(WrrHelper helper, Ticker ticker, Random ra
     this(new WrrHelper(OrcaOobUtil.newOrcaReportingHelper(helper)), ticker, random);
   }
 
+  @Override
+  protected Map<Object, ChildLbState> getChildLbMap(ResolvedAddresses resolvedAddresses) {
+    Map<Object, ChildLbState> childLbMap = new HashMap<>();
+    List<EquivalentAddressGroup> addresses = resolvedAddresses.getAddresses();
+    Object policyConfig = resolvedAddresses.getLoadBalancingPolicyConfig();
+    for (EquivalentAddressGroup eag : addresses) {
+      EquivalentAddressGroup key = stripAttrs(eag);
+      ChildLbState childLbState = new WeightedChildLbState(key, pickFirstLbProvider, policyConfig,
+          getInitialPicker());
+      childLbMap.put(key, childLbState);
+    }
+    return childLbMap;
+  }
```

Comment:
This almost exactly duplicates `MultiChildLoadBalancer.getChildLbMap()` - it just changes the type of the `ChildLbState` used. I don't think you want this kind of duplication. Did you consider other ways for subclasses to provide their own `ChildLbState` types? E.g. a factory method (say: `ChildLbState createChildLbState(eag, lbProvider, config, initialpicker)` in the base class that subclasses can override.

Reply:
The key for ClusterManager is not the eag, but rather a string retrieved from xds, so we do need something that converts ResolvedAddresses to the map (and why the generic map can't use eag as the key).   However, it is a good point that we can eliminate WRR implementing this logic by providing a createChildLbState.  And ... done.